runners/flink/src/main/java/org/apache/beam/runners/flink/CreateStreamingFlinkView.java

```diff
@@ -49,9 +56,20 @@ public CreateStreamingFlinkView(PCollectionView<ViewT> view) {
 
   @Override
   public PCollection<ElemT> expand(PCollection<ElemT> input) {
-    input
-        .apply(Combine.globally(new Concatenate<ElemT>()).withoutDefaults())
-        .apply(CreateFlinkPCollectionView.of(view));
+    PCollection<List<ElemT>> iterable;
+    // See https://github.com/apache/beam/pull/25940
+    if (view.getViewFn() instanceof PCollectionViews.IsSingletonView) {
+      TypeDescriptor<ElemT> inputType = input.getTypeDescriptor();
+      Preconditions.checkStateNotNull(inputType);
```

Comment:
```suggestion
      TypeDescriptor<ElemT> inputType =
          Preconditions.checkStateNotNull(input.getTypeDescriptor());
```
nit: this makes nullable check happy.